### PR TITLE
fix: email update

### PR DESCRIPTION
**Pull request by ignaciosantise**
# Changes
- fix: use `updateEmail` response to decide whether to redirect to first or second OTP screen

Steps to reproduce issue in production:
1. Create a NEW email account using an unused email
2. After connecting, change your email to a new one fast
3. Magic will send an OTP code directly to the new email and not to the current one
4. This causes the view to hang waiting for UPDATE_EMAIL_SUCCESS event

# Associated Issues
https://linear.app/walletconnect/issue/W3M-171/%5Bbug%5D-freeze-update-email-flow-when-otp-is-sent-to-new-email-only

### Related
Secure site PR 120

### CODE REVIEW

**Review by enesozturk — APPROVED** [review]

**Review by tomiir — APPROVED** [review]
This looks OK, just wondering if zod will break when secure site starts returning new data and old versions can't handle it. 

Did you try new secure site with older versions of w3m?

**Comment by ignaciosantise** [comment]
@tomiir yes, i tested the new secure site with an old version of w3m and it doesn't break because `zod.parse` removes the extra data. But i'll test everything again to be 100% sure.

**Comment by ignaciosantise** [comment]
@arein @lukaisailovic added update email tests

File: apps/laboratory/tests/shared/pages/ModalPage.ts [hunk 1]
```diff
@@ -226,4 +228,51 @@ export class ModalPage {
...
+
+    // Clear messages before putting new email
+    await email.deleteAllMessages(this.emailAddress)
+    await this.page.getByTestId('wui-email-input').locator('input').press('Enter')
+
+    // Wait until the next screen appears
+    await expect(this.page.getByText('Enter the code we sent')).toBeVisible({
+      timeout: 10_000
+    })
+    const confirmCurrentEmail = await this.page.getByText(this.emailAddress).isVisible()
```
Comment (arein) [inline apps/laboratory/tests/shared/pages/ModalPage.ts:249]: nitpick: would be nice to replace this with getById

Comment (ignaciosantise) [reply apps/laboratory/tests/shared/pages/ModalPage.ts:249]: in this case, i need to verify if the screen contains the current email or the new email. Getting the component by ID will always return true

File: apps/laboratory/tests/email.spec.ts [hunk 2]
```diff
@@ -2,6 +2,11 @@ import { expect } from '@playwright/test'
 import { testMEmail } from './shared/fixtures/w3m-email-fixture'
 import { SECURE_WEBSITE_URL } from './shared/constants'
 
+const mailsacApiKey = process.env['MAILSAC_API_KEY']
+if (!mailsacApiKey) {
+  throw new Error('MAILSAC_API_KEY is not set')
+}
+
```
Comment (lukaisailovic) [inline apps/laboratory/tests/email.spec.ts:9]: I don't  think we need this

Comment (ignaciosantise) [reply apps/laboratory/tests/email.spec.ts:9]: you mean to remove the if condition? In that case, i should set a default value for the key

i need the mailsacApiKey

File: apps/laboratory/tests/shared/pages/ModalPage.ts [hunk 3]
```diff
@@ -93,7 +96,8 @@ export class ModalPage {
     if (otp.replace(' ', '').length !== 6) {
       otp = email.getOtpCodeFromBody(emailBody)
     }
-    await this.enterOTP(otp)
+    // Remove empy space in OTP code 111 111
+    await this.enterOTP(otp.replace(' ', ''))
```
Comment (tomiir) [inline apps/laboratory/tests/shared/pages/ModalPage.ts:100]: maybe the `replace` could be in `getOtpCodeFromBody` to prevent repeating

Comment (ignaciosantise) [reply apps/laboratory/tests/shared/pages/ModalPage.ts]: done

**Review by tomiir — APPROVED** [review]

**Review by lukaisailovic — APPROVED** [review]

**Pull request merged**